dubbo-metadata/dubbo-metadata-api/src/main/java/org/apache/dubbo/metadata/MetadataInfo.java

```diff
@@ -339,7 +339,10 @@ public String getMethodParameter(String method, String key, String defaultValue)
         }
 
         private String getMethodParameter(String method, String key, Map<String, Map<String, String>> map) {
-            Map<String, String> keyMap = map.get(method);
+            Map<String, String> keyMap = null;
+            if (CollectionUtils.isNotEmptyMap(map)) {
+                keyMap = map.get(method);
+            }
```

Comment:
This check can avoid NPE from happing by checking before use, however, I don't think it can resolve the root  `methodParams` initialization issue as it has a race condition that needs to be resolved.